pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/pubsub/DSMtasksSubscription.java

```diff
@@ -85,21 +86,27 @@ public static void subscribeDSMtasks(String projectId, String subscriptionId) {
                     case PARTICIPANT_REGISTERED:
                         generateStudyDefaultValues(consumer, attributesMap);
                         break;
+                    case FILE_UPLOADED:
+                        sendFileUploadedNotification(notificationUtil,
+                                message.getAttributesOrDefault(PubSubAttributes.FILE_NAME.getValue(), null),
+                                message.getAttributesOrDefault(PubSubAttributes.USER_NAME.getValue(), null));
+                        consumer.ack();
+                        break;
```

Comment:
I am not seeing in existing DSM code that they are handling duplicate messages. Can you verify?
In case not familiar with issue: https://cloud.google.com/blog/products/data-analytics/handling-duplicate-data-in-streaming-pipeline-using-pubsub-dataflow
You can take look at DSS housekeeping handling of incoming code. I believe that de-duplication of messages is done by keeping track of message ids in a MySQL table.

Reply:
Well, on our side yes. PubSub guarantees at least one delivery, so on the DSM side it might happen that message will be processed twice. We actually can introduce a new table with the file ids that were processed and check if there is a record when receiving another message with.

Reply:
Let's move it to another task?